### DISCUSSION TITLE
Variable(s) to assess N limitation

### DISCUSSION THREAD

**Discussion by aswann** [ESCOMP/CTSM#1963] [Q&A] [answered]
Is there an appropriate variable or multiple variables to look at to find a change in N limitation between two experiments? I see a boost in LAI and NBP after I increase soil NO fluxes to the atmosphere in CAM6-CHEM w/BGC, and I'm trying to figure out if there is a real reason why this should happen. Thanks!

**Comment by wwieder** [comment, marked as answer]
Sorry, I'm not really sure what experiments you're looking at, but for a quick look I typically use FPI, which is metric of the soil biogeochemical N limitation (and easier to ncview; 1 = no limitation).  Otherwise I think @rosiealice recommends looking at the ratio between NPP_NUPTAKE/(NPP_NUPTAKE + NPP), alternatively  using GPP may also be appropriate?  This gives a proxy N limitation by normalizing FUN autotrophic respiration fluxes as a fraction of primary production.

**Reply by rosiealice** [reply]
Yes, NPP_NUPTAKE/(NPP_NUPTAKE + NPP) is the fraction of available C we are spending on N acquisition, as opposed to growth. Also the ratio with GPP would likely work fine too.

**Reply by aswann** [reply]
Thanks @wwieder and @rosiealice ! Super helpful.

For more context these are runs I made myself where I altered the soil NO flux emissions input file. I don't think that the land would know anything about these emissions to the atmosphere, other than possibly through higher N deposition. But I'm finding it *really* hard to track what is happening near the surface. The output variable from CAM called SFNO (surface flux of NO) does not match the input files, but I can't figure out in the code where there are any changes to surface fluxes. I expect that folks here won't know how to track this down specifically, but if you know where I could get some help figuring it out I'm all ears!

**Reply by wwieder** [reply]
Maybe contributors to CAM-CHEM can help with how FSNO fluxes in the coupled model, @mvalmartin or @lkemmons?

**Reply by lkemmons** [reply]
The output variable SFNO includes all of the NO emissions, including anthropogenic and biomass burning, which are also specified in atm_in (srf_emis_specifier) with 'NO -> {filename}'.

**Reply by aswann** [reply]
Yes, thanks @lkemmons! I have checked SFNO against the sum of all NO emissions (biomass burning, anthropogenic, and other (soils)) and they do not match. First they are off by a mostly constant factor (after converting to the same units). Second, they differ slightly beyond that. 

Here is what I wrote over [on the CESM discussion forums](https://bb.cgd.ucar.edu/cesm/threads/how-is-surface-flux-output-variable-sfno-related-to-input-files.8075/) - I was trying not to cross-post but this is relevant here as well!

> I accounted for the unit conversion and I summed the three components of NO flux. I still find the SFNO is bigger than the input flux such that I have to multiply the output flux by ~0.235 to make it equivalent to the input flux. Is there a canopy reduction factor being applied in reverse to the output flux (i.e. removing the effect of canopy deposition that was assumed in the input files)? 
> 
> The input and output are also offset by 15 days, and not identical even if there is a constant scaling between the two. I am guessing that this is because one is an input and one an output, so the output is reported at the beginning of the month and represents an average value over the month while the input is probably an instantaneous flux for middle day of the month. But are there other processes that I should be aware of?

**Reply by aswann** [reply]
![Unknown-15](https://user-images.githubusercontent.com/5891858/224092320-393d64b6-1e5e-40de-954a-cb54c4172b98.png)

**Reply by lkemmons** [reply]
The input files are in molecules/cm2/s and are output in kg(NO)/m2/s, using a molecular weight of 30 g/mole.
The input files have dates (I think) of the 15th of each month, and the model interpolates between the dates in the file at each timestep in the model. So, the monthly average in the output will be shifted.  I can't think of any other reasons for the difference between input and output.

**Reply by aswann** [reply]
I found an error and they are matching more closely now without an offset, but still with slight differences probably due to averaging  on the different time steps.

**Reply by lkemmons** [reply]
OK. Good.  I hadn't seen this correction and just checked one of my runs that indeed input and output are very similar:
Input: date, anthro, bb, soil, total
  20100116      92.2      13.2       7.4     112.8
  20100216      90.4       7.9       6.6     104.9
  20100316      90.0      10.0       8.3     108.3
  20100416      89.2       5.5      10.4     105.0
  20100516      88.4       6.4      13.3     108.1
  20100616      87.9      13.3      14.1     115.3
Output:
    20100201      112.120
    20100301      106.322
    20100401      107.458
    20100501      105.802
    20100601      108.696
    20100701      115.513
(remember that in monthly average files the date is at the end of the averaging period, so '20100701' is the June 2010 average.)

**Reply by aswann** [reply]
Although the discrepancy is smaller my input and output still aren't equivalent, and the error is ~20% of the total because the number is low. The SFNO is always larger which makes me think that averaging in time still can't explain this.
![Unknown-20](https://user-images.githubusercontent.com/5891858/224455254-dcb80acc-915d-4edc-931e-a57315ea214b.png)

**Reply by lkemmons** [reply]
Further discussion on the mismatch of input emissions and SFNO will be continued in https://bb.cgd.ucar.edu/cesm/threads/how-is-surface-flux-output-variable-sfno-related-to-input-files.8075/.

**Comment by mvalmartin** [comment]
Hi,  N deposition is coupled in the model, NHx and NOy (wet and dry deposition) can be passed from CAM to CLM, but you need to activate that in user_nl_cam as ndep_list = 'noy', 'nhx', otherwise CLM  uses prescribed N deposition.     @lkemmons should be able to confirm that, as N dep may be active now by default in newer released versions!

**Reply by lkemmons** [reply]
CAM-chem compsets in CESM2 should by default include nitrogen deposition from CAM to CLM.

**Reply by aswann** [reply]
I can also confirm that it looks like there is a response in FPI when I modify soil flux of NO, so I'm assuming this extra NO is causing extra deposition which is alleviating N limitation.

**Reply by rgknox** [reply]
@aswann , I'm not so up to date with the plant uptake diagnostics on the big-leaf CLM side (since I work with FATES), but,  I also use fraction of potential immobilzation (FPI) as one of my metrics to get a sense of N limitations.

**Comment by mvertens** [comment]
@mvalmartin - once the CAM PR https://github.com/ESCOMP/CAM/pull/735 is merged (it has been approved but I'm not sure why it has not been merged yet) - then CAM will always be passing NDEP to CLM. However, CLM in turn will need to be modified to always listen to the NDEP being passed in.

**Reply by mvalmartin** [reply]
Thanks for the clarification. So, this means that CLM is not using N deposition from CAM, as it is.  Am I correct? I have passed N dep from CAM to CLM in some of my runs, but i always had to tell CLM to use it.  Sorry if my note is confusing!

**Comment by mvertens** [comment]
@mvalmartin - sorry - I think my message was confusing. Currently CAM sends NDEP to CAM if it computes it prognostically. This is done via the settings in the drv_flds_in file that CAM produces and that CAM reads. If CAM does not produce the NDEP fields prognostically - then CLM is able to recognize this from the drv_flds_in file and read in NDEP itself. Once CAM always sends NDEP - then CLM can ignore the drv_flds_in namelist and always get NDEP data from CAM or DATM (which always sends NDEP right now). 
The key point is that once NDEP is always sent to CLM (particularly if the atm component is CAM) then CLM needs to turn off the logic where it decides if it will read in NDEP and just use what is passed. I hope that clarifies things.
